libs/ui/packages/native/src/components/cta/Button/index.tsx

```diff
@@ -116,12 +118,15 @@ const ButtonContainer = (props: ButtonProps & { hide?: boolean }): React.ReactEl
   const theme = useTheme();
   const { text } = getButtonColorStyle(theme.colors, props);
 
-  const IconNode = useMemo(
-    () =>
+  const IconNode = useMemo(() => {
+    if (isNewIcon) {
+      return Icon;
+    }
+    return (
       (iconName && <IconComponent name={iconName} size={ctaIconSize[size]} color={text.color} />) ||
-      (Icon && <Icon size={ctaIconSize[size]} color={text.color} />),
-    [iconName, size, Icon, text.color],
-  );
+      (typeof Icon === "function" && <Icon size={ctaIconSize[size]} color={text.color} />)
```

Comment:
Why is this needed ? (and the cast below)

Reply:
Tbh I tried conditional props without success
```
type CondtionalProps =
  | { isNewIcon: true; Icon: JSX.Element }
  | { isNewIcon?: false; Icon?: IconType };

export type ButtonProps = TouchableOpacityProps &
  BaseStyledProps & {
    onPressWhenDisabled?: TouchableOpacityProps["onPress"];
    iconName?: string;
    type?: "main" | "shade" | "error" | "color" | "default";
    size?: "small" | "medium" | "large";
    iconPosition?: "right" | "left";
    outline?: boolean;
    disabled?: boolean;
    pressed?: boolean;
    children?: React.ReactNode;
    pending?: boolean;
    displayContentWhenPending?: boolean;
    buttonTestId?: string;
  } & CondtionalProps;
```
Then I have the following TS error in the Button in LLM:
```
Types of property 'isNewIcon' are incompatible.
        Type 'boolean | undefined' is not assignable to type 'false | undefined'.
```
I will try to see if I can't find another fix.

Reply:
Discussed in DMs